embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/capture/crash/CrashDataSourceImpl.kt

```diff
@@ -0,0 +1,131 @@
+package io.embrace.android.embracesdk.capture.crash
+
+import io.embrace.android.embracesdk.Severity
+import io.embrace.android.embracesdk.anr.AnrService
+import io.embrace.android.embracesdk.arch.datasource.LogDataSourceImpl
+import io.embrace.android.embracesdk.arch.destination.LogEventData
+import io.embrace.android.embracesdk.arch.destination.LogWriter
+import io.embrace.android.embracesdk.arch.limits.OnOffLimitStrategy
+import io.embrace.android.embracesdk.arch.schema.EmbType
+import io.embrace.android.embracesdk.arch.schema.EmbType.System.ReactNativeCrash.embAndroidReactNativeCrashJsExceptions
+import io.embrace.android.embracesdk.arch.schema.SchemaType
+import io.embrace.android.embracesdk.arch.schema.TelemetryAttributes
+import io.embrace.android.embracesdk.config.ConfigService
+import io.embrace.android.embracesdk.gating.GatingService
+import io.embrace.android.embracesdk.internal.crash.CrashFileMarker
+import io.embrace.android.embracesdk.internal.utils.Uuid.getEmbUuid
+import io.embrace.android.embracesdk.logging.InternalEmbraceLogger
+import io.embrace.android.embracesdk.ndk.NdkService
+import io.embrace.android.embracesdk.opentelemetry.embAndroidThreads
+import io.embrace.android.embracesdk.opentelemetry.embCrashNumber
+import io.embrace.android.embracesdk.opentelemetry.exceptionMessage
+import io.embrace.android.embracesdk.opentelemetry.exceptionStacktrace
+import io.embrace.android.embracesdk.opentelemetry.exceptionType
+import io.embrace.android.embracesdk.opentelemetry.logRecordUid
+import io.embrace.android.embracesdk.payload.JsException
+import io.embrace.android.embracesdk.payload.LegacyExceptionInfo
+import io.embrace.android.embracesdk.prefs.PreferencesService
+import io.embrace.android.embracesdk.session.orchestrator.SessionOrchestrator
+import io.embrace.android.embracesdk.session.properties.EmbraceSessionProperties
+
+/**
+ * Intercept and track uncaught Android Runtime exceptions
+ */
+internal class CrashDataSourceImpl(
+    configService: ConfigService,
+    private val sessionOrchestrator: SessionOrchestrator,
+    private val sessionProperties: EmbraceSessionProperties,
+    private val anrService: AnrService?,
+    private val ndkService: NdkService,
+    @Suppress("UnusedPrivateMember") private val gatingService: GatingService,
```

Comment:
We need to retain similar behavior for gating on any crashes that are captured via OTel

Reply:
Yeap. We'll do this in a separate PR. The goal of this one is just to add the ability to switch between both crash payloads.